core/src/test/java/apoc/periodic/PeriodicTest.java

```diff
@@ -420,6 +423,38 @@ public void testIterateQueryFail() {
         testFail(query);
     }
 
+    @Test
+    public void testTruncate() {
+        db.executeTransactionally("UNWIND range(1,99999) AS x CREATE (:One{name:'Person_'+x})-[:FOO {id: x}]->(:Two {surname: 'Two'+x})<-[:BAR {idBar: x}]-(:Three {other: x+'Three'})");
+
+        db.executeTransactionally("CREATE INDEX ON :One(name)");
+        db.executeTransactionally("CREATE CONSTRAINT ON (a:Two) ASSERT a.surname IS UNIQUE");
+        db.executeTransactionally("CREATE INDEX ON :Three(other)");
+        db.executeTransactionally("CREATE CONSTRAINT ON (a:Actor) ASSERT a.name IS UNIQUE");
+
+        final int expectedNodes = 99999 * 3;
+        final int expectedRels = 99999 * 2;
+        assertCountEntitiesAndIndexes(expectedNodes, expectedRels, 4,2);
+
+        TestUtil.testCallEmpty(db, "CALL apoc.periodic.truncate", Collections.emptyMap());
+        assertCountEntitiesAndIndexes(0, 0, 4,2);
+
+        db.executeTransactionally("UNWIND range(1,99999) AS x CREATE (:One{name:'Person_'+x})-[:FOO {id: x}]->(:Two {surname: 'Two'+x})<-[:BAR {idBar: x}]-(:Three {other: x+'Three'})");
```

Comment:
nit: I'd move the `dropSchema:true` scenario to a separate test, in order to make sure that someone does not introduce a regression such that `dropSchema:true` would e.g. only drop the schema and not the data (which the current test would not catch).